src/oc_erchef/apps/data_collector/src/data_collector_http.erl

```diff
@@ -16,8 +16,19 @@
 
 -define(DEFAULT_HEADERS, [{"Content-Type", "application/json"}]).
 
+default_headers() ->
+    maybe_add_data_collector_token_header(?DEFAULT_HEADERS).
+
+maybe_add_data_collector_token_header(Headers) ->
+    case data_collector:token() of
+        undefined ->
+            Headers;
+        Token ->
+            Headers ++ [{"x-data-collector-token", Token}]
```

Comment:
Minor, but if we don't particularly care about the order: `[{"x-data-collector-token", Token} | Header]` is a bit cleaner I think.